### PR TITLE
Remove boston

### PR DESCRIPTION
**Pull request by rosecers**
As of 1.2, the boston dataset will no longer be supported by sklearn (https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html), so I have removed it from our examples and workflows.

### CODE REVIEW

File: tests/test_kernel_pcovr.py [hunk 1]
```diff
@@ -24,7 +24,12 @@ def __init__(self, *args, **kwargs):
 
         self.error_tol = 1e-6
 
-        self.X, self.Y = load_boston(return_X_y=True)
+        self.X, self.Y = get_dataset(return_X_y=True)
+
+        # for the sake of expedience, only use a subset of the dataset
+        idx = np.random.choice(len(self.X), 1000)
```
Comment (agoscinski) [inline tests/test_kernel_pcovr.py:30]: we use above random_state, in this case one needs to use the state to do the random choice
```
self.random_state.choice(len(self.X), 1000)
```
otherwise one could use np.random.seed before(0), then this also works

File: examples/Selectors-Pipelines.ipynb [hunk 2]
```diff
@@ -132,11 +154,45 @@
...
+       "(442, 10)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "X.shape"
```
Comment (agoscinski) [inline examples/Selectors-Pipelines.ipynb:182]: I guess this line should be deleted? Or at least commented why the shape of X is important

**Comment by rosecers** [comment]
> One global question - why did you decide to replace the Boston dataset with the Diabetes dataset? As I understood from sklearn's article about why they removed the Boston dataset, they will now conventionally use the California housing dataset instead.

Hey Sergei! The California dataset is *huge*, so it would slow down tests considerably. The diabetes dataset is another regression set that will serve the same purpose but not slow things down.

**Comment by hurricane642** [comment]
> > One global question - why did you decide to replace the Boston dataset with the Diabetes dataset? As I understood from sklearn's article about why they removed the Boston dataset, they will now conventionally use the California housing dataset instead.
> 
> Hey Sergei! The California dataset is _huge_, so it would slow down tests considerably. The diabetes dataset is another regression set that will serve the same purpose but not slow things down.

Hm, ok, fair enough

**Review by hurricane642 — APPROVED** [review]

File: tests/test_kernel_pcovr.py (hunk 1 above)
Comment (rosecers) [reply tests/test_kernel_pcovr.py]: Remedied!
